cdap-data-fabric/src/main/java/co/cask/cdap/data2/nosql/NoSqlStructuredTableRegistry.java

```diff
@@ -110,17 +93,18 @@ public void initialize() throws IOException {
   public void registerSpecification(StructuredTableSpecification specification) throws TableAlreadyExistsException {
     LOG.debug("Registering table specification {}", specification);
     StructuredTableId tableId = specification.getTableId();
-    Transactionals.execute(
-      transactional,
-      context -> {
-        Table table = context.getDataset(ENTITY_REGISTRY, Collections.emptyMap());
-        byte[] serialized = table.get(getRowKeyBytes(tableId), SCHEMA_COL_BYTES);
-        if (serialized != null) {
-          throw new TableAlreadyExistsException(tableId);
-        }
-        serialized = Bytes.toBytes(GSON.toJson(specification));
-        table.put(getRowKeyBytes(tableId), SCHEMA_COL_BYTES, serialized);
-      }, TableAlreadyExistsException.class);
+    MetricsTable table = getRegistryTable();
+    try {
+      byte[] rowKeyBytes = getRowKeyBytes(tableId);
+      byte[] serialized = table.get(rowKeyBytes, SCHEMA_COL_BYTES);
+      if (serialized != null) {
+        throw new TableAlreadyExistsException(tableId);
```

Comment:
would you say it's fine if the existing spec is the same?

Reply:
We use the `registerSpecification` as a proxy for create table in the NoSQL case. 

In the SQL case, if the table already exists during a table creation, then `TableAlreadyExistsException` is thrown. To keep the same behavior in NoSQL during table creation, we need to throw exception from `registerSpecification` when spec exists for a table.

Reply:
ok